### PR TITLE
201 collapsable accordian

### PR DESCRIPTION
**Pull request by ZaraGlynn**

### CODE REVIEW

File: src/styles/components/metadata/editors.scss [hunk 1]
```diff
@@ -11,20 +11,31 @@
...
     }
     .title {
-
+      &:hover {
+        cursor: pointer
+      }
     }
     .status {
-
+      &:hover {
```
Comment (Dan-Wood) [inline src/styles/components/metadata/editors.scss]: If you're putting this on each column, why not just put it on the row instead?

File: src/components/metadata/Items.tsx [hunk 2]
```diff
@@ -128,7 +130,7 @@ class ItemsClass extends React.Component<Props, State> {
     // Load item
     this.loadItem(s3Key);
 
-    this.setState({ items: {...this.state.items, ...items} } );
+    this.setState({isNewItem: true, items: {...this.state.items, ...items} } );
```
Comment (Dan-Wood) [inline src/components/metadata/Items.tsx]: Is this supposed to be on the items?

File: src/components/metadata/Items.tsx [hunk 3]
```diff
@@ -242,7 +244,7 @@ class ItemsClass extends React.Component<Props, State> {
         }
         {
           Object.entries(this.state.items).map( ( [s3Key, item] ) => {
-            return <ItemsDisplay isContributorPath={this.isContributorPath} key={s3Key} s3Key={s3Key} item={item} callback={this.fileUploadCallback} removeItem={this.props.allowRemoveItem ? this.removeItem : undefined} />;
+            return <ItemsDisplay isNewItem={!!this.state.isNewItem} isContributorPath={this.isContributorPath} key={s3Key} s3Key={s3Key} item={item} callback={this.fileUploadCallback} removeItem={this.props.allowRemoveItem ? this.removeItem : undefined} />;
```
Comment (Dan-Wood) [inline src/components/metadata/Items.tsx]: Are double bangs needed?

File: src/components/metadata/Items.tsx [hunk 4]
```diff
@@ -64,7 +65,8 @@ class ItemsClass extends React.Component<Props, State> {
     this._isMounted = false;
 
     this.state = {
-      items: {}
+      items: {},
+      isNewItem: false
```
Comment (Dan-Wood) [inline src/components/metadata/Items.tsx]: It's option but you're setting it?

File: src/components/admin/pages/items/Items.tsx [hunk 5]
```diff
@@ -103,6 +104,7 @@ class Items extends React.Component<RouteComponentProps, State> {
   async componentDidMount() {
     this._isMounted = true;
     this.isContributorPath = (this.props.location.pathname.match(/contributor/i));
+    this.isAdmin = (this.props.location.pathname.match(/admin/i));
```
Comment (Dan-Wood) [inline src/components/admin/pages/items/Items.tsx]: isAdminPath?

File: src/components/admin/pages/items/Items.tsx [hunk 6]
```diff
@@ -287,6 +289,7 @@ class Items extends React.Component<RouteComponentProps, State> {
             {
               typeof this.state.itemIndex !== 'undefined' && this.state.itemIndex >= 0 ?
                 <ItemEditor
+                  isAdmin={this.isAdmin}
                   isContributorPath={context.authorisation.hasOwnProperty('admin') ? false : this.isContributorPath}
```
Comment (Dan-Wood) [inline src/components/admin/pages/items/Items.tsx]: Why not just use context?

File: src/components/metadata/ItemEditor.tsx [hunk 7]
```diff
@@ -2654,19 +2654,19 @@ function withCollapse <P extends WithCollapseProps>(WrappedComponent: React.Comp
       return (
         <>
           <Row className="itemDetails" >
-            <Col className="itemIcons" sm="1"  onClick={this.toggleCollapse}>
+            <Col className="itemIcons" xs="1" sm="1"  onClick={this.toggleCollapse}>
```
Comment (Dan-Wood) [inline src/components/metadata/ItemEditor.tsx]: If it's 1 overall, you can just set xs='1'

File: src/components/metadata/ItemEditor.tsx [hunk 8]
```diff
@@ -2654,19 +2654,19 @@ function withCollapse <P extends WithCollapseProps>(WrappedComponent: React.Comp
...
-            <Col className="creators" onClick={this.toggleCollapse} xs="2" sm="3">
-              {this.state.item.creators ? this.state.item.creators : <></>}
+            <Col className="creators" onClick={this.toggleCollapse} xs="4" sm="3">
+              {this.state.item.creators ? this.state.item.creators.join(', ') : <></>}
             </Col>
             <Col className="status" onClick={this.toggleCollapse} xs="2" sm="2">
               {this.state.item.status ? <FaCheck color="green" size={25} /> : <FaTimes color="red" size={25} />}
             </Col>
-            <Col className="removeButton" sm="1">
+            <Col className="removeButton" xs="1" sm="1">
```
Comment (Dan-Wood) [inline src/components/metadata/ItemEditor.tsx]: Same here.

File: src/styles/components/metadata/editors.scss [hunk 9]
```diff
@@ -11,28 +11,35 @@
 
 .collectionEditor {
   .itemDetails {
-
-    margin-top: 50px;
+    border-bottom: 1px solid darkgrey;
```
Comment (Dan-Wood) [inline src/styles/components/metadata/editors.scss]: Can we use the variables? darkgrey isn't used anywhere.

File: src/styles/components/metadata/editors.scss [hunk 10]
```diff
@@ -11,28 +11,35 @@
...
+    margin-top: 10px;
     .creators {
+      margin-top: 20px;
```
Comment (Dan-Wood) [inline src/styles/components/metadata/editors.scss]: Again, instead of targeting each column, why not just the row?

File: src/styles/components/metadata/itemEditor.scss [hunk 11]
```diff
@@ -19,15 +21,9 @@
     position: absolute;
     height: 100%;
     width: 100%;
-    background: rgba(0,0,0,0.4);
+    background: rgba(0,0,0,0.2);
     opacity: 0;
     z-index: 0;
-    transition: opacity 1s ease-in;
```
Comment (Dan-Wood) [inline src/styles/components/metadata/itemEditor.scss]: Why was this removed?

File: src/styles/components/metadata/editors.scss (hunk 9 above)
Comment (Dan-Wood) [reply src/styles/components/metadata/editors.scss]: Can you make it border-top.

File: src/styles/components/metadata/itemEditor.scss (hunk 11 above)
Comment (ZaraGlynn) [reply src/styles/components/metadata/itemEditor.scss]: Becauseause it doesnt need to transition in, the overlay does that. Then it fades out to soft grey that you can't click

Comment (Dan-Wood) [reply src/styles/components/metadata/itemEditor.scss]: This is the overlay?

File: src/styles/components/metadata/editors.scss [hunk 12]
```diff
@@ -1,10 +1,49 @@
 .itemEditor, .collectionEditor {
   position: relative;
   color: #000;
+  width: 100%;
```
Comment (Dan-Wood) [inline src/styles/components/metadata/editors.scss]: Just a note, this is still happening

image

File: src/styles/components/metadata/editors.scss [hunk 13]
```diff
@@ -1,10 +1,49 @@
...
+
+.collectionEditor {
+  .itemDetails {
+    border-bottom: 1px solid darkgrey;
+    margin-top: 10px;
+    .creators {
+      margin-top: 20px;
+      word-break: break-word;
+      &:hover {
+        cursor: pointer
```
Comment (Dan-Wood) [inline src/styles/components/metadata/editors.scss]: These don't work on the 

http://localhost:3000/contributor/items/add

page.

File: src/styles/components/metadata/editors.scss (hunk 9 above)
Comment (Dan-Wood) [reply src/styles/components/metadata/editors.scss]: This page doesn't have the border styling http://localhost:3000/contributor/items/add

File: src/components/metadata/Items.tsx [hunk 14]
```diff
@@ -226,11 +230,29 @@ class ItemsClass extends React.Component<Props, State> {
...
+                Status
+              </Col>
+            </Row>
+            :
+            <></>
+        }
         {
           Object.entries(this.state.items).map( ( [s3Key, item] ) => {
-            return <ItemsDisplay isContributorPath={this.isContributorPath} key={s3Key} s3Key={s3Key} item={item} callback={this.fileUploadCallback} removeItem={this.props.allowRemoveItem ? this.removeItem : undefined} />;
+            return <ItemsDisplay isNewItem={!!this.state.isNewItem} isAdmin={!!this.props.isAdmin} isContributorPath={this.isContributorPath} key={s3Key} s3Key={s3Key} item={item} callback={this.fileUploadCallback} removeItem={this.props.allowRemoveItem ? this.removeItem : undefined} />;
```
Comment (Dan-Wood) [inline src/components/metadata/Items.tsx]: When I add an existing item in a new collection (http://localhost:3000/contributor/collections/add) it opens the accordion.

File: src/components/admin/pages/items/Items.tsx [hunk 15]
```diff
@@ -16,6 +16,7 @@ import 'react-bootstrap-table-next/dist/react-bootstrap-table2.min.css';
 import 'react-bootstrap-table2-paginator/dist/react-bootstrap-table2-paginator.min.css';
 
 import 'styles/components/admin/tables/modal.scss';
+import 'styles/components/metadata/editors.scss';
```
Comment (Dan-Wood) [inline src/components/admin/pages/items/Items.tsx]: Wrong place?

File: src/components/metadata/ItemEditor.tsx [hunk 16]
```diff
@@ -2654,19 +2654,19 @@ function withCollapse <P extends WithCollapseProps>(WrappedComponent: React.Comp
       return (
         <>
           <Row className="itemDetails" >
-            <Col className="itemIcons" xs="1" sm="1"  onClick={this.toggleCollapse}>
+            <Col className="itemIcons" sm="1"  onClick={this.toggleCollapse}>
```
Comment (Dan-Wood) [inline src/components/metadata/ItemEditor.tsx]: Removed wrong one.

File: src/styles/components/metadata/itemUpload.scss [hunk 17]
```diff
@@ -1,24 +1,18 @@
 @import '../../base/variables';
 
-.itemDetailsAdd {
-  margin-bottom: 20px;
+.accordianHeadings {
```
Comment (Dan-Wood) [inline src/styles/components/metadata/itemUpload.scss]: itemUpload.scss  != accordionCollapse.scss
